server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java

```diff
@@ -144,6 +144,10 @@ public Mappings getMappings(String user, Domain domain, int mappingLimit) throws
                         // Check if the returned mapping is the same as the
                         // input. If so return null to avoid loops
                         if (userName.equalsIgnoreCase(user) && targetDomain.equals(domain)) {
+                            if (type.equals(Type.Forward)) {
```

Comment:
oh yes, let's add a if and a continue in this very simple method :trollface:

Reply:
I did not find other solutions, sorry :trollface: 

Get a try, I think I spent 2 or 3 days on that. It is not easy as it seems.